esp32-wroom-rp/src/spi.rs

```diff
@@ -271,27 +284,28 @@ where
         self.wait_for_byte(ControlByte::Start as u8)
     }
 
-    fn read_and_check_byte(&mut self, check_byte: u8) -> Result<bool, self::Error> {
+    fn read_and_check_byte(&mut self, check_byte: &u8) -> Result<bool, self::Error> {
         match self.get_byte() {
             Ok(byte_out) => {
-                return Ok(byte_out == check_byte);
+                // Question: does comparing two &u8s work the way we would think?
+                return Ok(&byte_out == check_byte);
             }
             Err(e) => {
                 return Err(e);
             }
         }
     }
 
-    fn send_param<P: NinaParam>(&mut self, mut param: P) -> Result<(), self::Error> {
-        self.send_param_length(&mut param)?;
+    fn send_param<P: NinaParam>(&mut self, param: &P) -> Result<(), self::Error> {
+        self.send_param_length(param)?;
 
-        for byte in param.data().into_iter() {
-            self.bus.transfer(&mut [*byte]).ok().unwrap();
+        for byte in param.data().iter() {
+            self.bus.transfer(&mut [byte.clone()]).ok().unwrap();
```

Comment:
Take a look at the Clippy warning here...I think we could get rid of the clone and take advantage of built in Copy.

Reply:
I may be confused but it looks like this is already resolved on my copy of the branch. Maybe it was added as part of the work you did and I rebased on recently?